docs/private_net_tutorial.rst

```diff
@@ -0,0 +1,165 @@
+Raiden on Private Network Tutorial
+##################################
+
+Introduction
+============
+
+This tutorial shows how to run Raiden on a private network, using the ``master`` branch (this is useful when you are working on a pull-request).  This tutorial assumes Ubuntu 18.04.2 LTS and ``bash``.
+
+Creating a Virtual Environment
+==============================
+
+In a shell, run
+
+.. code:: bash
+
+ $ sudo apt-get install libncurses5-dev
+ $ mkdir priv_chain
+
+If ``mkdir`` fails, choose a different name, or move the existing ``priv_chain`` directory to somewhere else.
+
+.. code:: bash
+
+ $ cd priv_chain
+ $ virtualenv -p python3.7 env
+ $ source env/bin/activate
+
+You should now be in the virtual environment, where all Python package installations are separately managed.
+
+Now the command prompt should look like:
+
+.. code:: bash
+
+ (env) $
+
+
+Install Raiden and dependencies
+===============================
+
+.. code:: bash
+
+ (env) $ pwd
+ <snip>/priv_chain
+ (env) $ git clone https://github.com/raiden-network/raiden
+ (env) $ cd raiden
+ (env) $ pip install -r requirements.txt -c constraints.txt -e .
+
+Launch a private network
+========================
+
+Installing Geth
+---------------
+
+Follow `the guide <https://geth.ethereum.org/install-and-build/Installing-Geth>`__ and install Geth. A command ``geth`` should be available in your shell. This guide assumes version 1.8.26, but other versions might work.
+
+Preparing a genesis config
+--------------------------
+
+Prepare a file ``genesis.json`` with the following content (`@offerm <https://github.com/offerm>`__ kindly allowed to use his file here).
+
+.. code:: bash
+
+ (env) $ cd ..
+ (env) $ pwd
+ <snip>/priv_chain
+ (env) $ cat genesis.json
+ {
+ "config": {
+ "chainId": 4321,
+ "homesteadBlock": 0,
+ "eip150Block": 0,
+ "eip155Block": 0,
+ "eip158Block": 0,
+ "ByzantiumBlock": 0
+ },
+ "alloc": {},
+ "difficulty" : "0x1",
+ "gasLimit"   : "0x9880000"
+ }
+
+Starting a chain
+----------------
+
+With the ``genesis.json`` you can initialize a blockchain.
+
+.. code:: bash
+
+ (env) $ pwd
+ <snip>/priv_chain
+ (env) $ geth --datadir blkchain1 init genesis.json
```

Comment:
We have tooling for this, [`config_builder.py`](https://github.com/raiden-network/raiden/blob/4e90dbc804cc5015acb7a5f83511b9ecd6987b47/tools/config_builder.py) and [`startcluster.py`](https://github.com/raiden-network/raiden/blob/4e90dbc804cc5015acb7a5f83511b9ecd6987b47/tools/startcluster.py) which is a single command instead of this multi step guide, please refer to the tool help for info on how to use it (It looks like it needs to be updated though)

Reply:
These scripts enter the list of things that I'm interested in.  The list usually only gets longer unfortunately.